cmd/kwil-admin/nodecfg/generate.go

```diff
@@ -546,7 +556,7 @@ func incrementPort(incoming string, amt int) (string, error) {
 	// Reconstruct and return the new URL
 	newUrl := net.JoinHostPort(host, strconv.Itoa(port))
 
-	if res.Scheme != "localhost" {
+	if schemaExists {
```

Comment:
Nice.  I like this solution way better.  The scope of the changes logically should be limited to nodecfg (other than exporting `HasScheme`).
Will test in the a.m., but LGTM.